app/controllers/likes_controller.rb

```diff
@@ -1,8 +1,8 @@
 class LikesController < ApplicationController
   def save_like
-    @like = Like.new(post_id: params[:post_id], user_id: current_user.id)
     @post_id = params[:post_id]
-    existing_like = Like.find_by(post_id: params[:post_id], user_id: current_user.id)
+    @like = Like.new(post_id: @post_id, user: current_user)
```

Comment:
as much as possible, you want to query using the instance rather than the value.  Since you have already did the query for `current_user` you want to use it and query the `Like` by the `current_user`